### PR TITLE
fix: NaviViewItem might remain in over state when auto dismiss on click

### PR DESCRIPTION
**Pull request by dr1rrb**
closes https://github.com/unoplatform/uno/issues/7327
closes https://github.com/unoplatform/uno/issues/11610

## Bugfix
`NavigationViewItem` might remain in over state when auto dismiss on click

## What is the current behavior?
When an element is clicked and the `NavigationView` self dismiss itself, we receive a `OnPointerCancelled` (as the item is no longer in the visual tree), but it is being ignored as not "in bounds", so we are not clearing the private `isOver` flag.

## What is the new behavior?
We no longer ignore it.

### <samp>🤖 Generated by Copilot at 00b0bdf</samp>

Simplify pointer cancellation logic in `NavigationViewItem` to avoid stuck pressed state. This is part of a larger fix for pointer events and visual states in `NavigationView`.

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs [hunk 1]
```diff
@@ -1106,14 +1106,7 @@ private void ProcessPointerCanceled(PointerRoutedEventArgs args)
...
 			m_isPressed = false;
-			// m_isPointerOver should be true before this event so this doesn't need to be set to true in the else block...
-			// What this flag tracks is complicated because of the NavigationView sub items and the m_capturedPointers that are being tracked..
-			// We do this check because PointerCaptureLost can sometimes take the place of PointerReleased events.
-			// In these cases we need to test if the pointer is over the item to maintain the proper state.
-			if (IsOutOfControlBounds(args.GetCurrentPoint(this).Position))
-			{
-				m_isPointerOver = false;
-			}
+			m_isPointerOver = false;
```
Comment (dr1rrb) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: @MartinZikmund thoughts?

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: Was the comment ours?

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: It was coming from MUX, but maybe it was just to handle some UWP oddity

Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: The code is originally from WinUI sources.

https://github.com/microsoft/microsoft-ui-xaml/blob/210349d9d1e0aeb8ee144ec027071a68ac578bdb/dev/NavigationView/NavigationViewItem.cpp#L1077-L1085

I think we should comment the condition with "Uno docs" comment.

Comment (dr1rrb) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: It has been added 2 years ago https://github.com/microsoft/microsoft-ui-xaml/commit/c27a05caa0eeebaacdbd2106aebd12a6fc3dd912, most probably for the "Fix Animated chevron not working for heirarchical nav"

Comment (dr1rrb) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1109]: As it's from the WinUI repro, I'll restore the code and add a `#if false` to ignore it, and we will check in canaries.

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**